alacritty_terminal/src/grid/resize.rs

```diff
@@ -1,16 +1,21 @@
 //! Grid resize and reflow.
 
 use std::cmp::{min, Ordering};
+use std::mem;
 
+use crate::ansi::Color;
 use crate::index::{Column, Line};
-use crate::term::cell::Flags;
+use crate::term::cell::{Flags, ResetDiscriminant};
 
 use crate::grid::row::Row;
 use crate::grid::{Dimensions, Grid, GridCell};
 
-impl<T: GridCell + Default + PartialEq + Copy> Grid<T> {
+impl<T: ResetDiscriminant<Color> + GridCell + Default + PartialEq + Clone> Grid<T> {
     /// Resize the grid's width and/or height.
     pub fn resize(&mut self, reflow: bool, lines: Line, cols: Column) {
+        // Use empty template cell for resetting cells due to resize.
+        let template = mem::take(&mut self.cursor.template);
```

Comment:
This is the biggest drawback of the new approach I'd say. Since we just access the cursor template inside the grid automatically without a parameter this requires overwriting the template cell and then resetting it again later.

Generally this isn't that bad though I'd say and it makes sense since we want to switch the template context during reset, so it correctly represents what we're trying to do.

There are two alternatives, which both were significantly worse in my tests:

One could still take a `&T` as template everywhere. But since we pass these in from `term/mod.rs` we'd have to pass `&self.grid.cursor.template`, which obviously creates mutability problems. So we'd have to clone the cursor template just to pass a reference to it. Obviously not very nice.

One thing I have tested was passing a `template_override: Option<&T>` in as an argument. This allows replacing the template on demand when necessary with `Some(&T::default())`, but pass `None` whenever we want to just use the grid's template. Then we can just `template_override.unwrap_or(&self.cursor.template)` and everything's good to go. 

I personally think that approach would have been okay, but the additional parameters make it a bit more annoying to work with and somewhat strange. Ultimately the reason why I had to discard it was due to performance limitations. Since most of the time we do not have a template override, it's just wasteful to check. And that made a big performance difference.